cpp/ql/src/semmlecode.cpp.dbscheme

```diff
@@ -705,6 +705,11 @@ usertype_uuid(
     unique string uuid: string ref
 );
 
+mangled_type_name(
+    unique int id: @usertype ref,
+    string mangled_name : string ref
```

Comment:
Didn't we agree [on Jira](https://jira.semmle.com/browse/CPP-334?focusedCommentId=94651&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-94651) that there's no need to include the actual string for the mangled name in the db? All we need is a number that uniquely identifies it.

In any case, please take a good look at the performance and db size implications of this change for a project like range-v3 that may get huge mangled names due to template instantiations.

Reply:
I think that if we used IDs rather than strings, then we'd just move the data from the string pool to the ID pool, so there would be no space saving and we may as well have the strings available. We could hash the string to make it constant size, though. I'll get some numbers for all 3 options.

Reply:
Isn't the ID pool deleted after trap import unless some special flag is passed on the command line? I'd never expect the ID pool to be present in an exported snapshot, at least.

Reply:
Ah, fair point, I just checked and LGTM has exported (and hence sans idPool) snapshots.